src/PostgREST/App.hs

```diff
@@ -206,8 +204,13 @@ app dbStructure conf apiRequest =
         else return notFound
 
     (ActionRead, TargetRoot, Nothing) -> do
-      body <- encode <$> H.query schema accessibleTables
-      return $ responseLBS status200 [jsonH] $ cs body
+      let encodeApi ti = encodeOpenAPI ti host port
```

Comment:
I am not sure i follow, if you have a location like you mentioned with proxy_pass (that forwards requests to postgrest)
and someone requests the swagger schema, doesn't that response (json) contain the internal string 'host:port'? and then in the proxy, you would have to preprocess the response and only after that forward it to the user?